examples/gpt-2/main.cpp

```diff
@@ -703,11 +760,10 @@ bool gpt2_eval(
     ggml_allocr_alloc_graph(allocr, gf);
 
     // run the computation
-    struct ggml_cplan plan = ggml_graph_plan(gf, n_threads);
-    static std::vector<uint8_t> work_buffer;
-    work_buffer.resize(plan.work_size);
-    plan.work_data = work_buffer.data();
-    ggml_graph_compute(gf, &plan);
+    if (strcmp(ggml_backend_name(model.backend), "CPU") == 0) {
```

Comment:
minor : this should be more strongly-typed, such as `ggml_backend_is_cpu()`

Reply:
Yeah I think that's better than adding an enum. If you need to do a backend-specific operation, then using a backend-specific function to check if it is the correct backend is fine.